### PR TITLE
[IMP] resource: make working schedule optional in resource model

### PR DESCRIPTION
**Pull request by tivisse**
TaskID: 2984138

### CODE REVIEW

File: addons/resource/models/resource_resource.py [hunk 1]
```diff
@@ -118,10 +118,11 @@ def _adjust_to_calendar(self, start, end, compute_leaves=True):
                 start + relativedelta(hour=0, minute=0, second=0),
                 end + relativedelta(days=1, hour=0, minute=0, second=0),
             ]
-            calendar_start = resource.calendar_id._get_closest_work_time(start, resource=resource, search_range=search_range,
+            calendar = resource.calendar_id or resource.company_id.resource_calendar_id or self.env.company.resource_calendar_id
+            calendar_start = calendar._get_closest_work_time(start, resource=resource, search_range=search_range,
                                                                          compute_leaves=compute_leaves)
             search_range[0] = start
-            calendar_end = resource.calendar_id._get_closest_work_time(end if end > start else start, match_end=True,
+            calendar_end = calendar._get_closest_work_time(end if end > start else start, match_end=True,
```
Comment (MissingNoShiny) [inline addons/resource/models/resource_resource.py:125]:
```suggestion
            calendar_end = calendar._get_closest_work_time(max(start, end), match_end=True,
```

**Review by MissingNoShiny — COMMENTED** [review]
LGTM, thanks :slightly_smiling_face:

File: addons/mrp/models/mrp_workorder.py [hunk 2]
```diff
@@ -520,6 +520,8 @@ def _plan_workorder(self, replan=False):
         best_finished_date = datetime.max
         vals = {}
         for workcenter in workcenters:
+            if not workcenter.resource_calendar_id:
+                continue
```
Comment (Whenrow) [inline addons/mrp/models/mrp_workorder.py:524]: You can raise an error directly saying a calendar is mandatory to not raise the following one (the message is not the good one for that scenario) 
https://github.com/odoo/odoo/pull/110548/files#diff-fac872ffb03b811c4976eb2e52991ec544265332df814d92cfda658a5b917423R545

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**